File: src/UI/Mobile/MauiApp/Services/ProductService.cs
using EnterpriseTemplate.MauiApp.Services.Base;
using ProjectTemplate.Shared.Models;

namespace EnterpriseTemplate.MauiApp.Services;

public interface IProductService
{
    Task<PagedResponse<ProductResponseDto>?> GetProductsAsync(int page = 1, int pageSize = 10, string searchTerm = null, bool? isActive = null, CancellationToken cancellationToken = default);
}

public class ProductService : BaseService, IProductService
{
    public ProductService(HttpClient http) : base(http) { }

    public Task<PagedResponse<ProductResponseDto>?> GetProductsAsync(int page = 1, int pageSize = 10, string searchTerm = null, bool? isActive = null, CancellationToken cancellationToken = default)
    {
        var url = $"api/products?page={page}&pageSize={pageSize}";
        if (!string.IsNullOrEmpty(searchTerm)) url += $"&searchTerm={searchTerm}";
        if (isActive.HasValue) url += $"&isActive={isActive.Value}";
        
        return GetAsync<PagedResponse<ProductResponseDto>>(url, cancellationToken);
    }
}

File: src/UI/Web/Angular/src/app/core/services/product.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { PagedResponse, ProductResponse, CreateProductRequest } from '../../shared/models';

@Injectable({ providedIn: 'root' })
export class ProductService extends BaseService<ProductResponse, CreateProductRequest> {
  protected override entityPath = 'api/v1/Product';

  getProducts(page = 1, pageSize = 10, search = '', isActive?: boolean) {
    let url = `${this.fullUrl}?page=${page}&pageSize=${pageSize}`;
    if (search) url += `&searchTerm=${encodeURIComponent(search)}`;
    if (isActive !== undefined) url += `&isActive=${isActive}`;
    return this.http.get<PagedResponse<ProductResponse>>(url);
  }

  exportToExcel(search = '', isActive?: boolean) {
    let url = `${this.fullUrl}/ExportToExcel?`;
    if (search) url += `&searchTerm=${encodeURIComponent(search)}`;
    if (isActive !== undefined) url += `&isActive=${isActive}`;
    return this.http.get(url, { responseType: 'blob' });
  }
}
